applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp

```diff
@@ -822,39 +870,394 @@ void ShellThickElement3D4N::GetValueOnIntegrationPoints(const Variable<double>&
         const ProcessInfo& rCurrentProcessInfo)
 {
     SizeType size = GetGeometry().size();
-    if(rValues.size() != size)
+    if (rValues.size() != size) 
         rValues.resize(size);
 
-    std::vector<double> temp(size);
-
-    for(SizeType i = 0; i < size; i++)
-        mSections[i]->GetValue(rVariable, temp[i]);
-
-    const Matrix & shapeFunctions = GetGeometry().ShapeFunctionsValues();
-    Vector N(size);
-
-    for(SizeType i = 0; i < size; i++)
-    {
-        noalias(N) = row(shapeFunctions, i);
-        double& ival = rValues[i];
-        ival = 0.0;
-        for(SizeType j = 0; j < size; j++)
-        {
-            ival += N(j) * temp[j];
-        }
-    }
+	// The membrane formulation needs to iterate to find the correct 
+	// mid-surface strain values.
+	//
+	// Check if we are doing a non-linear analysis type. If not, print warning 
+	// for just the first element.
+
+    if (this->Id() == 1)
+	{
+		if (!rCurrentProcessInfo.Has(NL_ITERATION_NUMBER))
```

Comment:
@RiccardoRossi and @pooyan-dadvand what do you think abt this?
Can you think of a better solution?

Reply:
I will add the `KRATOS_WARNING_IF `soon

Reply:
I meant more how can the element know if the analysis type is linear or non-linear
The results can be plain wrong if a lin analysis is used for a non-lin element formulation

using the processinfo is just a workaround to know this

Reply:
Maybe we can use local flags in process info for this:
```c++
if (rCurrentProcessInfo.Is(ProcessInfo::SolutionType.NON_LINEAR&ProcessInfo::SolutionType.DYNAMIC)
```
Of course this should discussed in another issue!

Reply:
Created the issue here: #859

Reply:
:+1: